samples/gen-go-db/server/db/dynamodb/deployment.go

```diff
@@ -0,0 +1,568 @@
+					WriteCapacityUnits: aws.Int64(t.WriteCapacityUnits),
+				},
+			},
+			{
+				IndexName: aws.String("byEnvironment"),
+				Projection: &dynamodb.Projection{
+					ProjectionType: aws.String("ALL"),
+				},
+				KeySchema: []*dynamodb.KeySchemaElement{
+					{
+						AttributeName: aws.String("environment"),
+						KeyType:       aws.String(dynamodb.KeyTypeHash),
+					},
+					{
+						AttributeName: aws.String("date"),
+						KeyType:       aws.String(dynamodb.KeyTypeRange),
+					},
+				},
+				ProvisionedThroughput: &dynamodb.ProvisionedThroughput{
+					ReadCapacityUnits:  aws.Int64(t.ReadCapacityUnits),
+					WriteCapacityUnits: aws.Int64(t.WriteCapacityUnits),
+				},
+			},
+			{
+				IndexName: aws.String("byVersion"),
+				Projection: &dynamodb.Projection{
+					ProjectionType: aws.String("ALL"),
+				},
+				KeySchema: []*dynamodb.KeySchemaElement{
+					{
+						AttributeName: aws.String("version"),
+						KeyType:       aws.String(dynamodb.KeyTypeHash),
+					},
+				},
+				ProvisionedThroughput: &dynamodb.ProvisionedThroughput{
+					ReadCapacityUnits:  aws.Int64(t.ReadCapacityUnits),
+					WriteCapacityUnits: aws.Int64(t.WriteCapacityUnits),
+				},
+			},
+		},
+		ProvisionedThroughput: &dynamodb.ProvisionedThroughput{
+			ReadCapacityUnits:  aws.Int64(t.ReadCapacityUnits),
+			WriteCapacityUnits: aws.Int64(t.WriteCapacityUnits),
+		},
+		TableName: aws.String(t.name()),
+	}); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (t DeploymentTable) saveDeployment(ctx context.Context, m models.Deployment) error {
+	data, err := encodeDeployment(m)
+	if err != nil {
+		return err
+	}
+	_, err = t.DynamoDBAPI.PutItemWithContext(ctx, &dynamodb.PutItemInput{
+		TableName: aws.String(t.name()),
+		Item:      data,
+	})
+	return err
+}
+
+func (t DeploymentTable) getDeployment(ctx context.Context, environment string, application string, version string) (*models.Deployment, error) {
+	key, err := dynamodbattribute.MarshalMap(ddbDeploymentPrimaryKey{
+		EnvApp:  fmt.Sprintf("%s--%s", environment, application),
+		Version: version,
+	})
+	if err != nil {
+		return nil, err
+	}
+	res, err := t.DynamoDBAPI.GetItemWithContext(ctx, &dynamodb.GetItemInput{
+		Key:       key,
+		TableName: aws.String(t.name()),
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	if len(res.Item) == 0 {
+		return nil, db.ErrDeploymentNotFound{
+			Environment: environment,
+			Application: application,
+			Version:     version,
+		}
+	}
+
+	var m models.Deployment
+	if err := decodeDeployment(res.Item, &m); err != nil {
+		return nil, err
+	}
+
+	return &m, nil
+}
+
+func (t DeploymentTable) getDeploymentsByEnvAppAndVersion(ctx context.Context, input db.GetDeploymentsByEnvAppAndVersionInput, fn func(m *models.Deployment, lastDeployment bool) bool) error {
+	if input.VersionStartingAt != nil && input.StartingAfter != nil {
+		return fmt.Errorf("Must specify one of input.VersionStartingAt or input.StartingAfter")
+	}
```

Comment:
You should be able to specify neither, i.e. just query on the value of the hash key

Reply:
that's supported. This errors if both are specified.

Reply:
ahhh the error message is incorrect tho